pod/perlclass.pod

```diff
@@ -120,6 +120,27 @@ constructor at all. Expressions using C<//=> will also apply if the caller did
 pass the parameter but the value was undefined, and expressions using C<||=>
 will apply if the value was false.
 
+During a field initializing expression, the instance is not yet constructed
+and so the C<$self> lexical is not available.  However, the special
+C<__CLASS__> token may be used to obtain the name of the class being
+constructed, for example in order to invoke class methods on it to help in
+constructing values for fields.
+
+    class WithCustomField {
+        use constant DEFAULT_X => 10;
+        field $x = __CLASS__->DEFAULT_X;
+    }
+
+This allows subclasses to override the method with different behaviour.
+
+    class DifferentCustomField :isa(WithCustomField) {
+        sub DEFAULT_X { rand > 0.5 ? 20 : 30 }
+    }
```

Comment:
The initial design of Corinna was to separate methods and subroutines. Thus, you can't call subs like you can call methods. If we had the `:common` attribute, this could be written as:

```perl
    class WithCustomField {
        method DEFAULT_X :common {10 }
        field $x = __CLASS__->DEFAULT_X;
    }
```

This allows subclasses to override the method with different behaviour.

```perl
    class DifferentCustomField :isa(WithCustomField) {
        method DEFAULT_X :common { rand > 0.5 ? 20 : 30 }
    }
```

Reply:
Indeed so. This is the best we can write currently, but once we have common methods we can change the documentation to fit the better design.

Reply:
Maybe we should add a note about this in the documentation? If people start writing code relying on this, future releases will break it.

Reply:
It's all very new and experimental now anyway. Any of the docs should be read in that light anyway, so I don't think adding specific wording to basically every claim and sugestion is going to be all that helpful.